### DISCUSSION TITLE
Provide example combining react-easy-crop with react-uploady

### DISCUSSION THREAD

**Discussion by musicformellons** [rpldy/react-uploady#321] [Q&A] [answered]
I try to combine these two and run into problems. As [react-easy-crop](https://github.com/ricardo-ch/react-easy-crop) is quite popular (and provides dragging of the picture, instead of dragging the cropselection) such an example might be worthwhile. [Here](https://stackoverflow.com/questions/70975321/combine-react-easy-crop-with-react-uploady) is more info on the problems I run into, including a codesandbox.

**Comment by yoavniran** [comment, marked as answer]
Hey @musicformellons,  I provided an answer with a [working adaption (csb)](https://codesandbox.io/s/react-uploady-crop-and-upload-with-react-easy-crop-5g7vw) over on stackoverflow.
Turning this into a discussion for others to find in the future. 

Hope it helps

**Reply by musicformellons** [reply]
Really nice! Thank you very much for your quick help! I made the cropimage.js more in line with the 'original' and without the unnecessary rotation operations etc.

As my next requirement would be to have the image-crop be part of a form with validation I tried to combine it with the uploady form example, but I do not know how to trigger `onUploadCrop` from the form Submit button as they are part of different components. Also how to reference values to 'enable/disable' the submit button are unclear to me.  

What I have is this:
https://codesandbox.io/s/react-uploady-crop-and-upload-with-react-easy-crop-within-a-form-it114?file=/src/App.js

It should work as follows:

- submit button is disabled as form is invalid.
- press 'select file'
- selecting image, then it will show and can be cropped (or cancelled after which other image can be selected)
- other form fields should be filled
- press submit (which is 'enabled'): (validation then) 'onUploadCrop' will be triggered and cropped preview shown, other fields also uploaded.

Could you have a look maybe and suggest how to go about this?

**Reply by yoavniran** [reply]
This is a very interesting use-case. I will try and find time in the next few days to come up with an example. 

Without coding, I think it isnt trivial and may require a different approach. For example:

-  showing the crop based on a different event ([batch-add](https://react-uploady.netlify.app/docs/api/hooks/useBatchAddListener))
-  using `autoUpload=false` to prevent the file from starting to upload immediately. 
- save the crop result in some state
- replace the file with the cropped file using the [useRequestPreSend](https://react-uploady.netlify.app/docs/api/hooks/useRequestPreSend)

**Reply by musicformellons** [reply]
That would be great as I am still stuck on this!

**Reply by musicformellons** [reply]
I hope this is still on your todo-list?!

**Reply by yoavniran** [reply]
@musicformellons will try and get to it in the next few days 🤞

**Comment by yoavniran** [comment]
@musicformellons didnt forget about you. You should be able to follow the code in [this sandbox](https://codesandbox.io/s/react-uploady-crop-with-form-pvq4b) to get an example of cropping an image that will be uploaded as part of a form. Its simplified, of course, but should definitely help you get it working the way you need it.


https://user-images.githubusercontent.com/1102278/153894195-67766885-d4d6-4884-9644-bb05b5ac010b.mp4

**Reply by musicformellons** [reply]
@yoavniran Nice! Are you sure you send the right codesandbox, as it does not correspond to the .mp4 you attached (and the cropper in it does not work)?

**Reply by yoavniran** [reply]
not sure what you mean. Its the same sandbox and it works for me in multiple browsers...

**Reply by musicformellons** [reply]
Oops, yeah, it's fine. Will see whether I can turn it into the cropper I use! Thanks!
